haskell-src/Concordium/Types/Parameters.hs

```diff
@@ -27,42 +397,313 @@ import qualified Concordium.Crypto.SHA256 as Hash
+        supportsCooldownParametersAccessStructure AuthorizationsVersion1 = True
+
+        -- \|Whether time parameters are supported for an 'AuthorizationsVersion'.
+        supportsTimeParameters :: AuthorizationsVersion -> Bool
+        supportsTimeParameters AuthorizationsVersion0 = False
+        supportsTimeParameters AuthorizationsVersion1 = True
+
+        -- \|Parameter types that are conditionally supported at different 'ChainParametersVersion's.
+        data ParameterType
+            = -- \|Election difficulty (consensus parameter)
+              PTElectionDifficulty
+            | -- \|Time parameters
+              PTTimeParameters
+            | -- \|Mint rate per slot as part of mint distribution parameters
+              PTMintPerSlot
+            | -- \|Timeout parameters for V2 consensus (consensus parameter)
+              PTTimeoutParameters
+            | -- \|Minimum block time for V2 consensus (consensus parameter)
+              PTMinBlockTime
+            | -- \|Block energy limit for V2 consensus (consensus parameter)
+              PTBlockEnergyLimit
+            | -- \|Updatable cooldown parameters
+              PTCooldownParametersAccessStructure
+            | -- \|Finalization proof GAS rewards (GAS rewards parameter)
+              PTFinalizationProof
+
+        -- \|Whether a particular parameter is supported at a particular 'ChainParametersVersion'.
+        isSupported :: ParameterType -> ChainParametersVersion -> Bool
+        isSupported PTElectionDifficulty cpv = case consensusParametersVersionFor cpv of
+            ConsensusParametersVersion0 -> True
+            ConsensusParametersVersion1 -> False
+        isSupported PTTimeParameters cpv = supportsTimeParameters (authorizationsVersionFor cpv)
+        isSupported PTMintPerSlot cpv = supportsMintPerSlot (mintDistributionVersionFor cpv)
+        isSupported PTTimeoutParameters ChainParametersV0 = False
+        isSupported PTTimeoutParameters ChainParametersV1 = False
+        isSupported PTTimeoutParameters ChainParametersV2 = True
+        isSupported PTMinBlockTime ChainParametersV0 = False
+        isSupported PTMinBlockTime ChainParametersV1 = False
+        isSupported PTMinBlockTime ChainParametersV2 = True
+        isSupported PTBlockEnergyLimit ChainParametersV0 = False
+        isSupported PTBlockEnergyLimit ChainParametersV1 = False
+        isSupported PTBlockEnergyLimit ChainParametersV2 = True
+        isSupported PTCooldownParametersAccessStructure cpv = supportsCooldownParametersAccessStructure (authorizationsVersionFor cpv)
+        isSupported PTFinalizationProof ChainParametersV0 = True
+        isSupported PTFinalizationProof ChainParametersV1 = True
+        isSupported PTFinalizationProof ChainParametersV2 = False
+        |]
+ )
+
+-- |Constraint on a type level 'ParameterType' that can be used to get a corresponding
+-- 'SParameterType'. An alias for 'SingI'.
```

Comment:
I don't think it's useful to state that this is an alias for something. That's clear from the definition.